### DISCUSSION TITLE
Open file from :G or :Glog inline diff

### DISCUSSION THREAD

**Discussion by marlun** [tpope/vim-fugitive#1675] [Q&A] [answered]
If I do `:G` and get the inline diffs or if I do `:Glog` and open an inline diff is there a fast way to go to the file and the line where my cursor is on? I often want to see more context.

I've read https://github.com/tpope/vim-fugitive/issues/1230 and https://github.com/tpope/vim-fugitive/issues/1309 but just hitting `<CR>` is not working, not sure if they are not talking about the same thing I am or if something has changed?

**Comment by tpope** [comment, marked as answer]
`<CR>` is correct. Check `nmap <CR>` and make sure it looks like `:<C-U>exe <SNR>201_GF("edit")<CR>`. Note that in commit buffers, such as the ones loaded by `:Gclog`, it only works on `+` and `-` lines, as I didn't want to play favorites as to whether to jump to the old version or the new.

**Reply by marlun** [reply]
Looks correct with

    n  <CR>        &@:<C-U>exe <SNR>63_GF("edit")<CR>

Just allowing it for +/- lines seems like a good idea, I'm trying it on an added line but no luck.

**Reply by tpope** [reply]
What does "no luck" mean? I take it you don't end up at the desired destination b does the cursor move at all? Does the cursor blink?

What does `:echo fugitive#Cfile()` show in a commit buffer on a `+`/`-` line? (It won't work right in `:G` inline diffs, don't try it there.)

**Reply by marlun** [reply]
Sorry, maybe not the best explanation. It looked to me that nothing happend but the cursor do blink but stays at the same place.

I'm trying it on what opened up from `:Gclog` and the diff I'm trying looks like this:

    -    : 'text-grey-1 hover:bg-grey-95 cursor-pointer';
    +    : 'text-grey-2 hover:bg-grey-95 cursor-pointer';

And the output of `:echo fugitive#Cfile()` is:

    text-grey-2

With the cursor being on the `-` of text-grey-2.

**Reply by tpope** [reply]
This suggests it is failing to parse the filename out of the diff. This could be a weird setting in your `.gitconfig` changing how diffs are output (but not any setting I can think of), or it could be a weird setting in your vimrc fundamentally changing how parsing works - `'nomagic'`, `'isfname'`, and `'iskeyword'` spring to mind. Strip down your Vim config and Git config and try again.

**Reply by marlun** [reply]
For some reason I had `noprefix = true` in `.gitconfig` which caused the issue, thanks for the help! ❤️

**Reply by tpope** [reply]
Fugitive does guard against `noprefix` in some places (including the `:G` window) but wasn't doing so in commit buffers. I'll fix that.
